daemon/src/setup_maker.rs

```diff
@@ -0,0 +1,278 @@
+        (taker, confirm_order, taker_id): (
+            &(impl MessageChannel<maker_inc_connections::TakerMessage> + 'static),
+            &(impl MessageChannel<maker_inc_connections::ConfirmOrder> + 'static),
+            Identity,
+        ),
+        on_completed: &(impl MessageChannel<Completed> + 'static),
+        (on_stopping0, on_stopping1): (
+            &(impl MessageChannel<Stopping<Self>> + 'static),
+            &(impl MessageChannel<Stopping<Self>> + 'static),
+        ),
+    ) -> Self {
+        Self {
+            order,
+            quantity,
+            n_payouts,
+            oracle_pk,
+            announcement,
+            build_party_params: build_party_params.clone_channel(),
+            sign: sign.clone_channel(),
+            taker: taker.clone_channel(),
+            confirm_order: confirm_order.clone_channel(),
+            taker_id,
+            on_completed: on_completed.clone_channel(),
+            on_stopping: vec![on_stopping0.clone_channel(), on_stopping1.clone_channel()],
+            setup_msg_sender: None,
+        }
+    }
+
+    async fn contract_setup(&mut self, this: xtra::Address<Self>) -> Result<()> {
+        let order_id = self.order.id;
+        let cfd = Cfd::new(
+            self.order.clone(),
+            self.quantity,
+            CfdState::contract_setup(),
+        );
+
+        let (sender, receiver) = mpsc::unbounded();
+        // store the writing end to forward messages from the taker to
+        // the spawned contract setup task
+        self.setup_msg_sender = Some(sender);
+
+        let taker_id = self.taker_id;
+        let contract_future = setup_contract::new(
+            self.taker.sink().with(move |msg| {
+                future::ok(maker_inc_connections::TakerMessage {
+                    taker_id,
+                    msg: wire::MakerToTaker::Protocol { order_id, msg },
+                })
+            }),
+            receiver,
+            (self.oracle_pk, self.announcement.clone()),
+            SetupParams::new(
+                cfd.margin()?,
+                cfd.counterparty_margin()?,
+                cfd.order.price,
+                cfd.quantity_usd,
+                cfd.order.leverage,
+                cfd.refund_timelock_in_blocks(),
+            ),
+            self.build_party_params.clone_channel(),
+            self.sign.clone_channel(),
+            Role::Maker,
+            self.n_payouts,
+        );
+
+        spawn_fallible::<_, anyhow::Error>(async move {
+            let _ = match contract_future.await {
+                Ok(dlc) => this.send(SetupSucceeded { order_id, dlc }).await?,
+                Err(error) => this.send(SetupFailed { order_id, error }).await?,
+            };
+
+            Ok(())
+        });
+
+        Ok(())
+    }
+
+    async fn complete(&mut self, completed: Completed, ctx: &mut xtra::Context<Self>) {
+        let _ = self.on_completed.send(completed).await;
```

Comment:
We are ignoring the error here. I think we should either use `log_error` on the other side or at least not drop the `Result` here and log it :)

Reply:
I think you are right, but you may need to do the same in `collab_settlement_taker.rs`, since I copied this snippet from there.